include/seqan3/alignment/matrix/detail/combined_score_and_trace_matrix.hpp

```diff
@@ -0,0 +1,354 @@
+    //!\brief The reference type of the score matrix.
+    using score_matrix_reference_type = std::ranges::range_reference_t<score_matrix_t>;
+    //!\brief The reference type of the trace matrix.
+    using trace_matrix_reference_type = std::ranges::range_reference_t<trace_matrix_t>;
+
+    static_assert(std::ranges::viewable_range<score_matrix_reference_type>);
+    static_assert(std::ranges::viewable_range<trace_matrix_reference_type>);
+
+    //!\brief The combined column type.
+    using combined_column_type = decltype(views::zip(std::declval<score_matrix_reference_type>(),
+                                                     std::declval<trace_matrix_reference_type>()));
+    //!\brief The type of the score matrix iterator.
+    using score_matrix_iter_type = std::ranges::iterator_t<score_matrix_t>;
+    //!\brief The type of the trace matrix iterator.
+    using trace_matrix_iter_type = std::ranges::iterator_t<trace_matrix_t>;
+
+    // Befriend the base class.
+    template <std::ranges::input_range other_score_matrix_t, std::ranges::input_range other_trace_matrix_t>
+    //!\cond
+        requires (std::ranges::input_range<std::ranges::range_reference_t<other_score_matrix_t>> &&
+                  std::ranges::input_range<std::ranges::range_reference_t<other_trace_matrix_t>>)
+    //!\endcond
+    friend class combined_score_and_trace_matrix;
+
+    //!\brief The transform adaptor to convert the tuple from the zip view into a seqan3::detail::affine_cell_type.
+    static constexpr auto transform_to_combined_matrix_cell = std::views::transform([] (auto && tpl)
+        -> affine_cell_proxy<remove_cvref_t<decltype(tpl)>>
+    {
+        using fwd_tuple_t = decltype(tpl);
+        return affine_cell_proxy<remove_cvref_t<fwd_tuple_t>>{std::forward<fwd_tuple_t>(tpl)};
+    });
+
+    //!\brief The current iterator over the score matrix.
+    score_matrix_iter_type score_matrix_it;
+    //!\brief The current iterator over the trace matrix.
+    trace_matrix_iter_type trace_matrix_it;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief The value type.
+    using value_type = decltype(std::declval<combined_column_type>() | transform_to_combined_matrix_cell);
+    //!\brief The reference type.
+    using reference = value_type;
+    //!\brief The pointer type.
+    using pointer = void;
+    //!\brief The difference type.
+    using difference_type = std::ptrdiff_t;
+    //!\brief The iterator concept.
+    using iterator_concept = std::input_iterator_tag;
+    //!\}
+
+    /*!\name Constructor, assignment and destructor
+     * \{
+     */
+    iterator() noexcept = default; //!< Defaulted.
+    iterator(iterator const &) noexcept = default; //!< Defaulted.
+    iterator(iterator &&) noexcept = default; //!< Defaulted.
+    iterator & operator=(iterator const &) noexcept = default; //!< Defaulted.
+    iterator & operator=(iterator &&) noexcept = default; //!< Defaulted.
+    ~iterator() = default; //!< Defaulted.
```

Comment:
Won't defaulted ctors be noexcept if they can? So you want to make sure that they are noexcept here?

Reply:
It should be not enforced. Good point. Copy'n'paste error.

Reply:
(same for sentinel)